src/renderers/dom/client/eventPlugins/SimpleEventPlugin.js

```diff
@@ -483,6 +483,10 @@ for (var type in topLevelEventsToDispatchConfig) {
 var ON_CLICK_KEY = keyOf({onClick: null});
 var onClickListeners = {};
 
+function getDictionaryKey(inst) {
+  return '.' + inst._rootNodeID;
+}
+
```

Comment:
It doesn't break existing behavior because there is no observable difference as to what we use for keys. We just need to store some stuff on an object for each ID.

It improves perf because of some weird way V8 treats numeric keys. When all keys are numeric (like IDs are), for some reason `delete` becomes slower. Maybe it is similar to how sparse arrays are deoptimized. However if keys are not numeric (which we force by adding a dot before the number) it appears to create hash tables for those objects which is exactly what we want.

In general relying on something like this is very brittle. It is just VM implementation detail and can change any day. Still, if we can easily fix it without affecting performance in other engines, I think it's worth doing like in this case.